internal/server/storage/drivers/driver_zfs.go

```diff
@@ -210,7 +217,7 @@ func (d *zfs) FillConfig() error {
 
 			d.config["size"] = fmt.Sprintf("%dGiB", defaultSize)
 		}
-	} else if filepath.IsAbs(d.config["source"]) {
+	} else if sliceAll(devices, func(device string) bool { return filepath.IsAbs(device) }) {
```

Comment:
That logic seems incorrect to me as someone feeding "blah,/dev/foo" will be treated as having provided a valid source and will have their pool use the `blah` zpool, completely ignoring the invalid /dev/foo having been passed in.

Instead I think we need to flip around the conditions, basically checking for `if len(devices) == 1 && !filepath.IsAbs(devices[0])` to go with the existing zpool.

And then have the `else` handle the case where devices are passed in, in which case we should ensure that all of them are absolute paths or return an error back to the user.

Reply:
Added a vdev type check at the top. Would that suffice?

Reply:
I don't see how the change above makes this any different.

If I pass `source=blah,/dev/foo`, parseSource() will give you a default type (stripe) and two devices.

The check here will fail as `blah` isn't an absolute path which then makes us hit the `else` case that will then use `blah` as the pool, completely ignoring the fact that the user passed two devices and that it's invalid.

Reply:
I've rearranged the blocks to align with feedback.